python/cudf/cudf/comm/gpuarrow.py

```diff
@@ -63,7 +59,7 @@ class GpuArrowNodeReader(object):
     def __init__(self, table, index):
         self._table = table
         self._field = table.schema[index]
-        self._series = array_to_series(table.column(index))
+        self._series = Series(column.as_column(table.column(index)))
```

Comment:
Need to double check that this isn't accidentally causing a Device --> Host --> Device as pointed out by @trxcllnt

Reply:
I forked your branch and am testing whether we can detect the presence of Arrow CudaBuffers automatically in `utils.buffers_from_pyarrow`.